### PR TITLE
Add Sparse Merkle Tree documentation

### PR DESCRIPTION
**Pull request by KyrylR**

### CODE REVIEW

**Review by dovgopoly — APPROVED** [review]
LGTM

File: docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md [hunk 1]
```diff
@@ -0,0 +1,596 @@
...
+
+<figure>
+    <img src={require("/static/img/docs/sparse-merkle-tree.png").default} alt=""/>
+    <figcaption style={{"text-align": "center"}}>SMT that contains three elements</figcaption>
+</figure>
+
+When a new element is added to the tree, the binary representation of the key is used to determine the correct placement within the tree. The following steps are taken:
+
+1. The tree is recursively navigated to find an "empty" slot (an `Empty` node) where the new `Leaf` node can be placed.
+2. The path is decided based on the element's key—left is chosen for a bit value of 0 and right for a bit value of 1.
```
Comment (Arvolear) [inline docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md]:
```suggestion
2. The path is decided based on the element's key, left is chosen for a bit value of 0 and right for a bit value of 1.
```

File: docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+3. If a `Leaf` node is encountered during the insertion process, it is pushed down, and `Middle` nodes are inserted in their stead until a bit that differs between the leaves is identified. This scenario is illustrated on the image above. when the differing bit at position 3 leads to the creation of a branch at the tree's third level.
```
Comment (Arvolear) [inline docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md]:
```suggestion
3. If a `Leaf` node is encountered during the insertion process, it is pushed down, and `Middle` nodes are inserted in their stead until a bit that differs between the leaves is identified. This scenario is illustrated in the image above. When the differing bit at position 3 leads to the creation of a branch at the tree's third level.
```

File: docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+function setMaxDepth(Bytes32SMT storage tree, uint256 maxDepth_) internal;
+```
+
+```solidity
+function setMaxDepth(AddressSMT storage tree, uint256 maxDepth_) internal;
+```
+
+#### Description
+
+This function sets the maximum depth of the Merkle Tree. It reverts if the provided height is less than or equal to the current Merkle Tree depth. Also, it cannot exceed 256 due to the limitations of the uint256 data type; depths greater than 256 are not feasible.
```
Comment (Arvolear) [inline docs/getting-started/guides/libs/data-structures/sparse-merkle-tree.md]: What about the practical stack limit?

File: static/img/docs/sparse-merkle-tree.png
Comment (Arvolear) [inline static/img/docs/sparse-merkle-tree.png:1]: Middle Node 3?

**Review by Arvolear — APPROVED** [review]
LFG!

**Pull request merged**